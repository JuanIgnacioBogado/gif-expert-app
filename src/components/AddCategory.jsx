import PropTypes from 'prop-types';

export const AddCategory = ({ onAddCategory }) => {
  const handleSubmit = event => {
    event.preventDefault();

    const value = event.target.categoryInput?.value;
    if (!value?.trim()) return;

    onAddCategory(value.trim());
    event.target.categoryInput.value = '';
  };

  return (
    <form onSubmit={handleSubmit} aria-label="form">
      <input
        autoFocus
        name="categoryInput"
        type="text"
        placeholder="Buscar gifs"
      />
    </form>
  );
};

AddCategory.propTypes = {
  onAddCategory: PropTypes.func.isRequired
};
