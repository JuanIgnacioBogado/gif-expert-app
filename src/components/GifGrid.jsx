import PropTypes from 'prop-types';

import { GifItem } from './';

import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifGrid = ({ category }) => {
  const { gifs, isLoading } = useFetchGifs(category);

  if (isLoading) return <h2>Cargando...</h2>;

  return (
    <>
      <h3>{category}</h3>

      <div className="card-grid">
        {gifs?.map(gif => (
          <GifItem
            {...{
              ...gif,
              key: gif.id
            }}
          />
        ))}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
};
