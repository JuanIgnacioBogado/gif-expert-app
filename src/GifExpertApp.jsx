import { useState } from 'react';
import { AddCategory, GifGrid } from './components';

export const GifExpertApp = () => {
  const [categories, setCategories] = useState(['One Punch']);

  const onAddCategory = category => {
    if (categories.includes(category)) return;
    setCategories([category, ...categories]);
  };

  return (
    <>
      <h1>GifExpertApp</h1>

      <AddCategory onAddCategory={onAddCategory} />

      {categories?.map(category => (
        <GifGrid key={category} category={category} />
      ))}
    </>
  );
};
