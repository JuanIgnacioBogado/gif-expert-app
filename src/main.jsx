import { StrictMode } from 'react';
import ReactDOM from 'react-dom/client';

import { GifExpertApp } from './GifExpertApp';

import './styles.css';

ReactDOM.createRoot(document.getElementById('root')).render(
  <StrictMode>
    <GifExpertApp />
  </StrictMode>
);
