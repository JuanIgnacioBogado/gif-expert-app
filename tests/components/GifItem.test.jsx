import { render, screen } from '@testing-library/react';

import { GifItem } from '../../src/components';

describe('GifItem', () => {
  const title = 'calamardo';
  const url = 'https://one-punch.com/saitama.jpg';

  test('should to match snapshot', () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test('should to show image with the correct url and alt', () => {
    render(<GifItem title={title} url={url} />);
    // screen.debug();
    // expect(screen.getByRole('img').src).toBe(url);
    // expect(screen.getByRole('img').alt).toBe(title);

    const { src, alt } = screen.getByRole('img');
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test('should to show title in the component', () => {
    render(<GifItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
