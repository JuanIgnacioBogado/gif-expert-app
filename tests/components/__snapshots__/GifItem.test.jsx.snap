// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItem should to match snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="calamardo"
      src="https://one-punch.com/saitama.jpg"
    />
    <p>
      calamardo
    </p>
  </div>
</div>
`;
